test/core.jl

```diff
@@ -5863,7 +5863,7 @@ let
     @test b.x === Int8(91)
     @test b.z === Int8(23)
     @test b.y === A23367((Int8(1), Int8(2), Int8(3), Int8(4), Int8(5), Int8(6), Int8(7)))
-    @test sizeof(b) == sizeof(Int) * 3
+    @test sizeof(b) == 12
```

Comment:
So I think this is more correct now? The change is via the call to `jl_islayout_inline` in `jl_f_sizeof` [here](https://github.com/JuliaLang/julia/blob/8d4f6d24c0fa87509fcf40788ca2c83cdfecd9af/src/builtins.c#L372).